src/geofabrics/dem.py

```diff
@@ -395,21 +445,28 @@ def __init__(
         raw_dem = raw_dem.squeeze("band", drop=True)
         self._write_netcdf_conventions_in_place(raw_dem, catchment_geometry.crs)
 
-        # Ensure all values outside the exents are nan as that defines the dense extents
-        # and clip the dense dem to the catchment extents to ensure performance
+        # Clip to catchment and set the data_source layer to NaN where there is no data
         raw_dem = raw_dem.rio.clip(catchment_geometry.catchment.geometry, drop=True)
-        raw_dem = raw_dem.rio.clip(extents.geometry, drop=False)
-
+        raw_dem["data_source"].data = numpy.where(
```

Comment:
The `where` method of xarray could be used here, to avoid accessing the underlying `.data` object, staying at the highest API when possible:
```
raw_dem["data_source"] =  raw_dem["data_source"].where(...)
```

Reply:
done